### ISSUE TITLE
`trace_block` returns additional array elem when comapred to infura

### ISSUE THREAD

**Issue by kamilchodola** [NethermindEth/nethermind#6401]
This is the JSON RPC call:
curl URL   -X POST   -H "Content-Type: application/json"   --data '{"jsonrpc": "2.0", "method": "trace_block", "params": ["0x11f47dd"], "id": 298340364307158711}'

When Executed on nethermind node (1.23.0 version) vs Infura public rpc on Nethermind side I got extra value in trace - below are responses from both:
Nethermind:
[trace_block_1230.json](https://github.com/NethermindEth/nethermind/files/13729174/trace_block_1230.json)

Infura:
[trace_block_infura.json](https://github.com/NethermindEth/nethermind/files/13729175/trace_block_infura.json)

According to online JSON compare tool it is about this section:
<img width="800" alt="image" src="https://github.com/NethermindEth/nethermind/assets/43241881/f1e8f32a-7c99-40ec-8691-661a849858ab">

**Comment by LukaszRozmej** [comment]
Yes we are adding additional reward to traces. This was required for Gnosis and coordinated to be same as erigon: https://github.com/NethermindEth/nethermind/issues/4616

**Comment by LukaszRozmej** [comment]
This is also controlled by `ParityTraceTypes.Rewards` parameter. We use `ParityTraceTypes.Trace | ParityTraceTypes.Rewards` for trace_block, maybe infura uses only `Trace`?

**Comment by kamilchodola** [comment]
@claude can you check from code if this was ever improved?
